File: Challenge/file-structure-template/coding/htb/solver.py
data = input().strip().split(" ")
T = int(data[0])
N = int(data[1])
nums = input().strip().split(" ")
nums = list(map(int, nums))

seen = {}
for num in nums:
    complement = T - num
    if complement in seen:
        pair = sorted((num, complement))
        print(f"{pair[0]} {pair[1]}")
        exit()
    seen[num] = True

File: Challenge/file-structure-template/coding/htb/solver.cc
#include <iostream>
#include <unordered_map>
#include <vector>
#include <algorithm>

int main() {
    int T, N;
    std::cin >> T >> N;
    std::vector<int> nums(N);
    
    for (int i = 0; i < N; ++i) {
        std::cin >> nums[i];
    }
    
    std::unordered_map<int, bool> seen;
    for (int i = 0; i < N; ++i) {
        int complement = T - nums[i];
        if (seen.count(complement)) {
            int a = nums[i];
            int b = complement;
            if (a > b) std::swap(a, b);
            std::cout << a << " " << b << std::endl;
            return 0;
        }
        seen[nums[i]] = true;
    }

    return 0;
}
